src/m365/spo/commands/group/group-member-add.ts

```diff
@@ -211,104 +202,55 @@ class SpoGroupMemberAddCommand extends SpoCommand {
 
     const validUserNames: string[] = [];
     const invalidUserNames: string[] = [];
-    const userIdentifiers: string = args.options.userName || args.options.email || args.options.aadGroupId! || args.options.aadGroupName! || args.options.userId!.toString();
+    const identifiers: string = args.options.userName || args.options.email || args.options.aadGroupId! || args.options.aadGroupName! || args.options.userId!.toString();
 
     return Promise
-      .all(userIdentifiers.split(',').map(async userIdentifier => {
-        const user = userIdentifier.trim();
+      .all(identifiers.split(',').map(async identifier => {
+        const trimmedIdentifier = identifier.trim();
         try {
           if (args.options.userId) {
-            await this.spoUserGet(args.options, user, logger, validUserNames);
+            if (this.verbose) {
+              logger.logToStderr(`Get UPN from SharePoint for user ${trimmedIdentifier}`);
+            }
+            const spoUser = await spo.getUserById(args.options.webUrl, trimmedIdentifier);
+            validUserNames.push(spoUser.UserPrincipalName);
+          }
+          else if (args.options.userName) {
+            validUserNames.push(trimmedIdentifier);
+          }
+          else if (args.options.aadGroupId) {
+            validUserNames.push(trimmedIdentifier);
           }
-          else if (args.options.aadGroupId || args.options.aadGroupName) {
-            await this.aadGroupGet(args.options, user, logger, validUserNames);
+          else if (args.options.aadGroupName) {
+            if (this.verbose) {
+              logger.logToStderr(`Get UPN from Azure AD for group ${trimmedIdentifier}`);
+            }
+            const group = await aadGroup.getGroupByDisplayName(trimmedIdentifier);
```

Comment:
We recently added a new util to retrieve only the group ID by it's name. Could you rebase to the latest main and use that new function?